### PR TITLE
Adding obsoletion tag

### PR DESCRIPTION
**Pull request by sabrinatoro**

### CODE REVIEW

File: src/sparql/reports/obsoletiontagcheck.sparql
Comment (twhetzel) [inline src/sparql/reports/obsoletiontagcheck.sparql:1]: @sabrinatoro did you intend to commit this SPARQL query?

File: src/sparql/reports/obsoletiontagcheck.sparql
Comment (twhetzel) [inline src/sparql/reports/obsoletiontagcheck.sparql:13]: If this query should be in the PR, where did these Mondo IDs come from?

**Review by twhetzel — CHANGES_REQUESTED** [review]
I usually check that the terms in the candiatesForObsoletion [gSheet](https://docs.google.com/spreadsheets/d/1tt1Wk70j9XiHLV1vKQyNiHhaazh286pobpJk1ecSCCg/edit?gid=1661957154#gid=1661957154) match-up with the changes in `mondo-edit.obo`. Depending on what the crossed out lines in the gSheet mean, there are at most ~34~ 32 new obsolete candidates, but many more in the `mondo-edit.obo` file.

File: src/sparql/reports/obsoletiontagcheck.sparql
Comment (sabrinatoro) [reply src/sparql/reports/obsoletiontagcheck.sparql:1]: Yes, I committed this sparql query because it is helpful to have when having a lot of obsoletion candidates and we want to check them before merging the ROBOT template (otherwise the ROBOT merging fail, one term at a time).

File: src/sparql/reports/obsoletiontagcheck.sparql
Comment (sabrinatoro) [reply src/sparql/reports/obsoletiontagcheck.sparql:13]: My goal was to have the query in the repo so we can use it later. If we want to use it again, we would need to replace the MONDO ids with the new obsoletion candidates. Because of this, I didn't think it was necessary to keep the 300+ terms in the query, and I removed them.

**Comment by sabrinatoro** [comment]
The number of obsoletion candidates is more than that this time. If you look at the [googlesheet](https://docs.google.com/spreadsheets/d/1tt1Wk70j9XiHLV1vKQyNiHhaazh286pobpJk1ecSCCg/edit?gid=434975340#gid=434975340): 
- Tab ROBOT_AddObsSubsetTag to add in release 2024-07 : 346 new obsoletion candidates
- Tab ROBOT_AddObsTagForMerge to add in release 2024-07 : 5 new obsoletion candidates
- Tab ROBOT_NewTermSplit-ObsTag to add in release 202407 : 1 new obsoletion candidate

**Comment by sabrinatoro** [comment]
WRT the sparql query: 
When adding a lot of obsoletion candidate tag, we run into failing on the ROBOT template merging because a comment already exist (for example). To facilitate this, I (with Nico's help) created a sparql query to identify the candidates which will make the ROBOT merging fail. 
I thought it would be a good tool to have in the repo for future use. 
I removed most of the MONDO:ID present in the original query because it doesn't make sense to keep them (the important and stable part of the query is everything else). 
I was in a hurry and did not explain everything related around this sparql query. Sorry. I hope it makes more sense now.

**Comment by twhetzel** [comment]
For "Tab ROBOT_AddObsSubsetTag to add in release 2024-07", should there be comments for all rows in the spreadsheet and therefore a `comment: Reason of obsoletion...` property on the Mondo class?

There are only 339 classes with `comment: Reason of obsoletion...`, but 352 classes with `subset: obsoletion_candidate`.

**Review by twhetzel — APPROVED** [review]
Thanks for pointing me to the other tabs. 

I added a comment about not all classes having a reason for obsoletion comment, but will approve in the meantime so this can be merged if it those missing properties were intended.

**Comment by sabrinatoro** [comment]
13 candidates already had a comment, so I had to add the obsoletion-related comment manually (otherwise the ROBOT merge would fail because terms have more than one comment). 
This is why you see 339 classes with comment: Reason of obsoletion..., but 352 classes with subset: obsoletion_candidate.

**Pull request merged**
